src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,148 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int CAPACITY_MULTIPLIER = 2;
+
+    private Node<K, V>[] table;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        int position = getPositionInTable(key);
+        Node<K, V> newNode = new Node<>(getKeyHashCode(key), key, value, null);
+
+        if (table[position] == null) {
+            table[position] = newNode;
+
+            size++;
+        } else {
+            Node<K, V> currElem = getElementByKeyOrLast(key, position);
 
+            if ((key == null && currElem.getKey() == null)
+                            || (currElem.getKey() != null && currElem.getKey().equals(key))
+            ) {
+                currElem.setValue(value);
+            } else {
+                currElem.setNext(newNode);
+
+                size++;
+            }
+        }
+
+        if (size >= table.length * LOAD_FACTOR) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        int position = getPositionInTable(key);
+
+        if (table[position] == null) {
+            return null;
+        }
+
+        Node<K, V> currElem = getElementByKeyOrLast(key, position);
+
+        return currElem.getValue();
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getPositionInTable(K key) {
+        return key == null ? 0 : getKeyHashCode(key) % table.length;
+    }
+
+    private Node<K, V> getElementByKeyOrLast(K key, int position) {
+        Node<K, V> currElem = table[position];
+
+        while (((currElem.getKey() != null && !currElem.getKey().equals(key))
+                || (currElem.getKey() == null && key != null))
+                && currElem.getNext() != null
+        ) {
+            currElem = currElem.getNext();
```

Comment:
```suggestion
        while (currElem != null ) {
        if(Objects.equals(currElem.key, key)) {
            currElem = currElem.getNext();
            return;
```
try to simplify this method using this construction (while conditional really complicated)